Xamarin.PropertyEditing.Mac/Controls/Custom/CharTextField.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using AppKit;
+using Foundation;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class CharTextField : NSTextField
+	{
+		string cachedValueString;
+		NSText CachedCurrentEditor { get; set; }
+
+		public event EventHandler ValidatedEditingEnded;
+
+		public override CoreGraphics.CGSize IntrinsicContentSize => new CoreGraphics.CGSize (30, 20);
+
+		public CharTextField ()
+		{
+			BackgroundColor = NSColor.Clear;
+			ControlSize = NSControlSize.Small;
+			StringValue = string.Empty;
+			TranslatesAutoresizingMaskIntoConstraints = false;
+		}
+
+		public override bool ShouldBeginEditing (NSText textObject)
+		{
+			CachedCurrentEditor = textObject;
+			cachedValueString = textObject.Value;
+			CachedCurrentEditor.Delegate = new CharValidationDelegate (this);
+			return true;
+		}
+
+		public virtual void NotifyValidatedEditingEnded ()
+		{
+			ValidatedEditingEnded?.Invoke (this, EventArgs.Empty);
+		}
+
+		public virtual void ResetInvalidInput ()
+		{
+			this.StringValue = cachedValueString;
+		}
+	}
+
+	public class CharValidationDelegate : NSTextViewDelegate
```

Comment:
```suggestion
	internal class CharValidationDelegate : NSTextViewDelegate
```

Reply:
Given the commonality here with other validation classes, we should consolidate most of this to a class that just has an override for whether the text validates or not, essentially line 61.